src/validation.h

```diff
@@ -508,6 +513,28 @@ class CChainState {
 
     bool LoadBlockIndex(const Consensus::Params& consensus_params, CBlockTreeDB& blocktree) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    /**
+     * Update the on-disk chain state.
+     * The caches and indexes are flushed depending on the mode we're called with
+     * if they're too large, if it's been a while since the last write,
+     * or always and in all cases if we're in prune mode and are deleting files.
+     *
+     * If FlushStateMode::NONE is used, then FlushStateToDisk(...) won't do anything
+     * besides checking if we need to prune.
+     */
+    bool FlushStateToDisk(
+        const CChainParams& chainparams,
+        CValidationState &state,
+        FlushStateMode mode,
+        int nManualPruneHeight = 0);
+
+    //! Unconditionally flush all changes to disk.
+    void ForceFlushStateToDisk();
+
+    //! Prune blockfiles from the disk if necessary and then flush chainstate changes
```

Comment:
Nit: flushing is done _before_ pruning. Rename to `FlushAndPrune()`?

Reply:
Ehh, well it depends on what you're talking about being flushed - i.e. `pcoinsTip` get flushed [after](https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L2191) the prune happens.

Reply:
I was talking about the chain state, which is the biggest thing :-)

Reply:
Yeah, `pcoinsTip` (chainstate) is flushed *after* pruning.

Reply:
Ah wait, never mind, you're right. Why do we flush the chainstate cache after pruning? Is that in order to minimize the chances of running out of disk space?